drivers/sensor/st/iis3dwb/iis3dwb.c

```diff
@@ -0,0 +1,453 @@
+/* ST Microelectronics IIS3DWB accelerometer senor
+ *
+ * Copyright (c) 2025 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Datasheet:
+ * https://www.st.com/resource/en/datasheet/iis3dwb.pdf
+ */
+
+#define DT_DRV_COMPAT st_iis3dwb
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/logging/log.h>
+
+#include "iis3dwb.h"
+#include "iis3dwb_rtio.h"
+
+LOG_MODULE_REGISTER(IIS3DWB, CONFIG_SENSOR_LOG_LEVEL);
+
+static int32_t iis3dwb_set_range_raw(const struct device *dev, uint8_t range)
+{
+	struct iis3dwb_data *data = dev->data;
+	const struct iis3dwb_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+
+	data->range = range;
+	return iis3dwb_xl_full_scale_set(ctx, range);
+}
+
+static int32_t iis3dwb_set_odr_raw(const struct device *dev, uint8_t odr)
+{
+	struct iis3dwb_data *data = dev->data;
+	const struct iis3dwb_config *cfg = dev->config;
+	stmdev_ctx_t *ctx = (stmdev_ctx_t *)&cfg->ctx;
+
+	data->odr = odr;
+	return iis3dwb_xl_data_rate_set(ctx, odr);
+}
+
+static int iis3dwb_odr_set(const struct device *dev,
+			    const struct sensor_value *val)
+{
+	iis3dwb_odr_xl_t odr;
+
+	switch (val->val1) {
+	case 0:
+		odr = IIS3DWB_XL_ODR_OFF;
+		break;
+
+	default:
+		odr = IIS3DWB_XL_ODR_26k7Hz;
```

Comment:
Other drivers only allow rounding up. We should keep that up, if the user asked for 25kHz then serving a higher value is OK, but if the user asks for 100kHz we should fail.